dev/bots/prepare_package.dart

```diff
@@ -292,10 +292,46 @@ class ArchiveCreator {
     _outputFile = File(path.join(outputDir.absolute.path, _archiveName));
     await _installMinGitIfNeeded();
     await _populateCaches();
+    await _validate();
     await _archiveFiles(_outputFile);
     return _outputFile;
   }
 
+  /// Validates the integrity of the release package.
+  ///
+  /// Currently only checks that macOS binaries are codesigned. Will throw a
+  /// [PreparePackageException] if the test failes.
+  Future<void> _validate() async {
+    // Only validate in strict mode, which means `--publish`
+    if (!strict || !platform.isMacOS) {
+      return;
+    }
+    // Validate that the dart binary is codesigned
+    final String dartPath = path.join(
```

Comment:
Are there any other binaries that we should check? e.g. `pub`?

Reply:
i'll add a TODO, my plans are to migrate this script into `//flutter/dev/tools`, then it can hook into [this logic](https://github.com/flutter/flutter/blob/master/dev/tools/lib/codesign.dart) to test every binary that is executable.

I tried to have this script call that one, but it was tricky, and this script is difficult to test--I was worried it would work locally but fail on LUCI when I'm actually trying to publish.

Reply:
That's a valid concern: this type of this is always hard to test without actually publishing. A TODO is fine.